stdlib/public/runtime/SymbolInfo.h

```diff
@@ -43,14 +43,74 @@ namespace swift {
 struct SymbolInfo {
 private:
 #if defined(_WIN32) && !defined(__CYGWIN__)
-  SYMBOL_INFO_PACKAGE _package;
+  const void *_symbolAddress;
+  const char *_symbolName;
+  const char *_moduleFileName;
+  const void *_moduleBaseAddress;
 #elif SWIFT_STDLIB_HAS_DLADDR
   Dl_info _info;
 #endif
 
+#if defined(_WIN32) && !defined(__CYGWIN__)
+  SymbolInfo(const void *symbolAddress,
+             const char *symbolName,
+             const char *moduleFileName,
+             const void *moduleBaseAddress)
+    : _symbolAddress(symbolAddress),
+      _symbolName(symbolName),
+      _moduleFileName(moduleFileName),
+      _moduleBaseAddress(moduleBaseAddress)
+  {}
+#endif
+
 public:
   SymbolInfo() {}
 
+#if defined(_WIN32) && !defined(__CYGWIN__)
+  SymbolInfo(const SymbolInfo &other) {
+    _symbolName = nullptr;
```

Comment:
You don't want to duplicate these?

Reply:
Ah, the duplication is done by the `*this = other` line just below. The reason for setting these to `nullptr` was to make sure we don't fall over by calling `free()` with uninitialised pointer values.